bazel/integration/test_runner/package_json.ts

```diff
@@ -49,10 +49,18 @@ export function updateMappingsForPackageJson(
 ): PackageJson {
   const newPackageJson = {...packageJson};
 
-  updateMappingForRecord(newPackageJson.dependencies ?? {}, mappings);
-  updateMappingForRecord(newPackageJson.devDependencies ?? {}, mappings);
-  updateMappingForRecord(newPackageJson.optionalDependencies ?? {}, mappings);
-  updateMappingForRecord(newPackageJson.resolutions ?? {}, mappings);
+  updateMappingForRecord(newPackageJson, 'dependencies', mappings);
+  updateMappingForRecord(newPackageJson, 'dependencies', mappings);
+  updateMappingForRecord(newPackageJson, 'optionalDependencies', mappings);
+  // The object for Yarn resolutions will not directly match with the mapping keys
+  // specified here, as resolutions usually follow a format as followed:
+  //  1. `**/<pkg-name>`
+  //  2. `<other-pkg>/**/<pkg-name>`
+  //  3. `<pkg-name>`
+  // More details here: https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/.
+  // We pass a regular expression which matches the `<pkg-name>` so that the mappings can
+  // be applied for resolutions as well.
+  updateMappingForRecord(newPackageJson, 'resolutions', mappings, /(?:^|\/)?([^/]+)$/);
```

Comment:
By making the `(?:^|\/)` optional, it basically makes it redundant. So, you could as well remove it altogether (and the regex would remain the same):

```suggestion
  updateMappingForRecord(newPackageJson, 'resolutions', mappings, /([^/]+)$/);
```